packages/app/src/containers/Canvas.tsx

```diff
@@ -249,6 +255,68 @@ export default memo(function Canvas() {
           event.preventDefault();
           break;
         }
+        case 'editPath': {
+          const layer: any = Selectors.getLayerAtPoint(
+            CanvasKit,
+            state,
+            insets,
+            rawPoint,
+            {
+              clickThroughGroups: false,
+              includeHiddenLayers: false,
+              includeArtboardLayers: false,
+            },
+          );
+          const boundingRects = getBoundingRectMap(
+            getCurrentPage(state),
+            state.selectedObjects,
+            {
+              clickThroughGroups: true,
+              includeArtboardLayers: false,
+              includeHiddenLayers: false,
+            },
+          );
+
+          if (!layer) return;
+
+          const boundingRect = boundingRects[layer.do_objectID];
+          //Why dose TS not recognize layer.points?
+          let selectedPoint: SelectedPoint | undefined = undefined;
+          layer.points.forEach((layer_point: any, index: number) => {
+            const decodedPoint = decodeCurvePoint(layer_point, boundingRect);
+            if (isPointInRange(decodedPoint.point, point)) {
+              selectedPoint = [layer.do_objectID, index];
+            }
+          });
+
+          if (selectedPoint) {
```

Comment:
I didn't realize Sketch allows deselecting the only point while holding shift (unlike for layers), so we can simplify as:

```ts
if (selectedPoint) {
  const alreadySelected = state.selectedPointLists[
    selectedPoint[0]
  ]?.includes(selectedPoint[1]);

  dispatch(
    'selectPoint',
    selectedPoint,
    event.shiftKey
      ? alreadySelected
        ? 'difference'
        : 'intersection'
      : 'replace',
  );
} else {
  dispatch('selectPoint', undefined);
}
```

Since I mostly wrote this bit I don't feel bad about rewriting it 😂